scipy/sparse/linalg/tests/test_pydata_sparse.py

```diff
@@ -0,0 +1,226 @@
+import pytest
+
+import numpy as np
+import scipy.sparse as sp
+import scipy.sparse.linalg as splin
+
+from numpy.testing import assert_allclose
+
+try:
+    import sparse
+except ImportError:
+    sparse = None
+
+pytestmark = pytest.mark.skipif(sparse is None,
+                                reason="pydata/sparse not installed")
+
+
+def test_isolve_gmres():
+    # Several of the iterative solvers use the same
+    # isolve.utils.make_system wrapper code, so test just one of them.
+    np.random.seed(1234)
+
+    A = sparse.COO(np.random.rand(5, 5))
+    b = np.random.rand(5)
+
+    x, info = splin.gmres(A, b, atol=1e-15)
+    assert info == 0
+    assert isinstance(x, np.ndarray)
+    assert_allclose(A @ x, b)
+
+
+def test_lsmr():
+    np.random.seed(1234)
+
+    A_dense = np.random.rand(5, 5)
+    A = sparse.COO(A_dense)
+    b = np.random.rand(5)
+
+    res0 = splin.lsmr(A_dense, b)
+    res = splin.lsmr(A, b)
+    assert_allclose(res[0], res0[0])
+
+
+def test_lsqr():
+    np.random.seed(1234)
+
+    A_dense = np.random.rand(5, 5)
+    A = sparse.COO(A_dense)
```

Comment:
Oh, this appears quite a number of times. Maybe a fixture `sparse_class` is in order. 😄

Reply:
done